pulseguardian/guardian.py

```diff
@@ -97,6 +123,21 @@ def update_queue_information(self, queue_data):
                 owner = None
             queue = Queue(name=q_name, owner=owner)
 
+        # add the queue bindings to the db.
+        bindings = pulse_management.queue_bindings(config.rabbit_vhost,
+                                                   queue.name)
+        for binding in bindings:
+            db_binding = Binding.query.filter(
+                Binding.exchange == binding["source"],
+                Binding.routing_key == binding["routing_key"]).first()
```

Comment:
Shouldn't you be filtering by queue as well? Looks like this will only create a binding if no other bindings with the same exchange & routing_key exist.